device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -429,4 +361,43 @@ public void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeC
         //Codes_SRS_DEVICE_IO_34_020: [This function shall register the callback with the transport.]
         this.transport.registerConnectionStatusChangeCallback(statusChangeCallback, callbackContext);
     }
+
+    @Override
+    public void onTransportConnectionClosed()
+    {
+        if (taskScheduler != null)
+        {
+            this.taskScheduler.shutdown();
+            this.taskScheduler = null;
+        }
+    }
+
+    @Override
+    public void onTransportConnectionOpened()
+    {
+        if (this.sendTask == null)
+        {
+            this.sendTask = new IotHubSendTask(this.transport);
+        }
+
+        if (this.receiveTask == null)
+        {
+            this.receiveTask = new IotHubReceiveTask(this.transport);
+        }
+
+        if (this.taskScheduler == null)
+        {
+            this.taskScheduler = Executors.newScheduledThreadPool(2);
+
```

Comment:
Can this be static? Why would you want to spawn new threads everytime transport is opened. Eg - What happens during Retry.

Reply:
When the connection is DISCONNECTED in the transport layer, it calls up to the onTransportConnectionLost() method in DeviceIO which stops sends the sender and receiver threads. This is necessary for the DeviceIO layer to be "closed" when the Transport layer is closed.

Once the client is fully closed, we expect customers to call open again, which will start up the send/receive threads again.

I can make the thread pool static, but the open()  must spawn the send/receive threads, and close() must shut them down.

As for DISCONNECTED_RETRYING happening in the transport layer, nothing happens to the DeviceIO layer as a result. The sender and receiver threads are left running.